ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/tests/ParticipantPruningIT.scala

```diff
@@ -0,0 +1,601 @@
+          cannotPruneBegin,
+          Status.Code.INVALID_ARGUMENT,
+          "prune_up_to needs to be absolute and not a boundary LEDGER_BEGIN")
+        assertGrpcError(
+          cannotPruneEnd,
+          Status.Code.INVALID_ARGUMENT,
+          "prune_up_to needs to be absolute and not a boundary LEDGER_END")
+      }
+  })
+
+  test(
+    "PRFailPruneByNonHexOffset",
+    "Pruning a participant specifying a non-hexadecimal offset should fail",
+    allocate(NoParties))(implicit ec => {
+    case Participants(Participant(participant)) =>
+      for {
+        cannotPruneNonHexOffset <- participant
+          .prune(LedgerOffset.of(LedgerOffset.Value.Absolute("cofefe")), 1)
+          .failed
+      } yield {
+        assertGrpcError(
+          cannotPruneNonHexOffset,
+          Status.Code.INVALID_ARGUMENT,
+          "prune_up_to needs to be a hexadecimal string and not")
+      }
+  })
+
+  test(
+    "PRFailPruneByOutOfBoundsOffset",
+    "Pruning a participant specifying an offset after the ledger end should fail",
+    allocate(NoParties))(implicit ec => {
+    case Participants(Participant(participant)) =>
+      for {
+        actualEndExclusive <- participant.currentEnd()
+        cannotPruneOffsetBeyondEnd <- participant
+          .prune(actualEndExclusive, 1)
+          .failed
+      } yield {
+        assertGrpcError(
+          cannotPruneOffsetBeyondEnd,
+          Status.Code.INVALID_ARGUMENT,
+          "prune_up_to needs to be before ledger end")
+      }
+  })
+
+  test("PRPruneTxTrees", "Prune succeeds as observed by transaction trees", allocate(SingleParty))(
+    implicit ec => {
+      case Participants(Participant(participant, submitter)) =>
+        for {
+          endOffsetAtTestStart <- participant.currentEnd()
+
+          size <- populateLedger(participant, submitter, batchesToPopulate)
+
+          txBeforePrune <- participant.transactionTrees(
+            participant
+              .getTransactionsRequest(parties = Seq(submitter), begin = endOffsetAtTestStart))
+
+          offsetToPruneUpTo = offsetAt[TransactionTree](txBeforePrune, itemsToPrune, _.offset)
+
+          offsetOfFirstSurvivingTransaction = offsetAt[TransactionTree](
+            txBeforePrune,
+            itemsToPrune + 1,
+            _.offset)
+
+          _ <- participant.prune(offsetToPruneUpTo)
+
+          txAfterPrune <- participant.transactionTrees(
+            participant
+              .getTransactionsRequest(Seq(submitter))
+              .update(_.begin := offsetToPruneUpTo))
+
+          cannotReadAnymore <- participant
+            .transactionTrees(
+              participant
+                .getTransactionsRequest(Seq(submitter))
+                .update(
+                  _.begin := offsetAt[TransactionTree](
+                    txBeforePrune,
+                    itemsToPrune - 1, // This offset is the largest exclusive offset we can no longer read from after
+                    _.offset))
```

Comment:
Could be factored out as a separate value, next to the related `offsetToPruneUpTo` and `offsetOfFirstSurvivingTransaction` values. Maybe as a `offsetOfLastRemovedTransaction`?